manifests/init.pp

```diff
@@ -1,4 +1,155 @@
+# @param config_template
+#   This determines which template puppet should use for the chrony configuration.
+# @param config_keys
+#   This sets the file to write chrony keys into.
+# @param config_keys_template
+#   This determines which template puppet should use for the chrony key file.
+# @param config_keys_owner
+#   Specify unix owner of chrony keys file, defaults to 0.
+# @param config_keys_group
+#   Specify unix group of chrony keys files, defaults to 0 on ArchLinux and chrony on Redhat.
+# @param config_keys_mode
+#   Specify unix mode of chrony keys files, defaults to 0644 on ArchLinux and 0640 on Redhat.
+# @param keys
+#   An array of key lines.  These are printed as-is into the chrony key file.
+# @param local_stratum
+#   Override the stratum of the server which will be reported to clients
+#   when the local reference is active. Defaults to 10.
+# @param stratumweight
+#   Sets how much distance should be added per stratum to the synchronisation distance when chronyd
+#   selects the synchronisation source from available sources.
+#   When not set, chronyd's default will be used, which since version 2.0 of chrony, is 0.001 seconds.
+# @param log_options
+#   Specify which information is to be logged.
+# @param package_ensure
+#   This can be set to 'present' or 'latest' or a specific version to choose the
+#   chrony package to be installed.
+# @param package_name
+#   This determines the name of the package to install.
+# @param peers
+#   This selects the servers to use for NTP peers (symmetric association).
+#   It is an array of servers.
+# @param servers
+#   This selects the servers to use for NTP servers.  It can be an array of servers
+#   or a hash of servers to their respective options.
+# @param refclocks
+#   This should be a Hash of hardware reference clock drivers to use.  They hash
+#   can either list a single list of options for the driver, or any array of
+#   multiple options if the same driver is used for multiple hardware clocks.
+#
+#   Example:
+#   ```puppet
+#   refclocks => { 'PPS' => [ '/dev/pps0 lock NMEA refid GPS',
+#                            '/dev/pps1:clear refid GPS2' ],
+#                  'SHM' => '0 offset 0.5 delay 0.2 refid NMEA noselect' }
+#   ```
+# @param makestep_seconds
+#   Together with `makestep_updates`, this configures the `makestep` parameter of `chronyd`.
+#   Usually, `chronyd` never steps the time, but applies a slew
+#   after the initial synchronization.
+#   This setting configures for how many _updates_ the time may be stepped
+#   if the adjustment is larger than specified _seconds_.
+#
+#   For virtual machines which are suspended and resumed for a prolonged time,
+#   stepping the time may be wanted. In this case, set `makestep_updates` to `-1`
+#   to allow stepping the time for any update.
+# @param makestep_updates
+#   Together with `makestep_seconds`, this configures the `makestep` parameter of `chronyd`.
```

Comment:
I wasn't sure what to do here.  Happy to change this to new wording (to remove the duplication), or perhaps this could be done in a future PR.

Reply:
Please reword this, it's horrible to get you head around these sentences.

Reply:
Reworded and linked to official docs (which aren't great either TBH).  I've also added an `@example` and data types to the parameters.